### DISCUSSION TITLE
Next steps

### DISCUSSION THREAD

**Discussion by DanyaoWang-001167576** [ian-tiggeman/T24S2T1EXTG08#7] [General]
Hi Ian,

I have completed the mortgage feature and committed it to my local feature branch. Since I see you haven’t pushed your feature branch to GitHub yet, does one of us merge into their local development branch first and push to the remote repo. Then the other pulls from the remote repo, and merges the other feature into their local development branch and pushes back to GitHub?

On page 9 of the Assignment document, under section d., it mentions “_Push the feature branch to the remote repository_” so I’m just a bit confused about what we need to do next.

If you have a clear understanding of the steps, please let me know.

Thanks,
Cara

**Comment by ian-tiggeman** [comment]
Hi Cara,
I know the instruction are very confusing.  This is what I think the next steps are:
1. We both Push our feature branches to GitHub.
2. You then merge your feature branch into the development branch (on your local PC).  Resolve any conflicts and Commit development branch.
3. You push the new development branch to GitHub (communication required before and after the merge)
4. I then pull the latest development branch from GitHub and merge my feature in, resolve and commit (as per you in section 2).
5. I test and push the new complete development branch code.

Just keep an eye on the screenshots we need to take along the way, as they are confusing and hidden (as usual).
This should complete section C.
We can then start section D...

or something like this.  Sound right?
(I will add your public key now)

Regards,
Ian

**Reply by DanyaoWang-001167576** [reply]
That sounds much clearer now. Thank you!
I’ve already highlighted two screenshot requirements that I don’t quite understand, lol. 
It’s getting late, so I’ll ask you about it tomorrow :)
Cara

**Reply by ian-tiggeman** [reply]
FYI, I have pushed my feature branch here to GitHub.
Try to do yours and see if you have trouble, if so we will probably need to sort out the SSH Key issue/confusion.

**Comment by ian-tiggeman** [comment]
No worries.
Perhaps a phone or Teams call tomorrow would help us sort out a few details?
Chat to you tomorrow.

________________________________

**Comment by DanyaoWang-001167576** [comment]
I'll let you know how it goes tomorrow. 🙂
________________________________

**Comment by DanyaoWang-001167576** [comment]
Hi Ian,
I pushed my feature branch to GitHub this morning without any issues (I didn’t need the key). I can merge the mortgage feature once you approve it.
Before merging, I have two questions about the screenshots:
    1.	_Push the feature branch to the remote repository. Capture a screenshot showing the revision graph with the checkout of the remote feature branch._
    2.	_Before the merge, show the code differences with the previous revision in the show log._
    
I have a screenshot for the first one, but I’m not sure if it's accurate.

![Screenshot 2024-09-07 101211](https://github.com/user-attachments/assets/c713c2d1-2513-4a2c-94c3-1effbfb1a98c)



For the second one, I don’t quite understand what it’s asking for. Is it asking for a screenshot from the show log? If so, how do I capture the code differences?
Let me know what you think.
Regards,
Cara

**Reply by ian-tiggeman** [reply]
Hi Cara,
You have my all clear to merge when you have you screen shots.
Your provided screenshot looks like you have the LOCAL feature branch checked out, not the REMOTE branch.  I believe you have to select checkout "remotes/T24S2T1EXTG08/feature_mortgage".
![image](https://github.com/user-attachments/assets/2660577a-626b-4ea5-acc7-627e7b1b8601)

Second one - after selecting "Show log" from the right click, then once you have the show log screen, double click a file in the lower window will show you the differences between current and previous version file.  It might be this?

**Reply by ian-tiggeman** [reply]
Example:
![image](https://github.com/user-attachments/assets/a3005d75-a7c5-484d-a62c-f9e2c0fcd685)

**Comment by DanyaoWang-001167576** [comment]
Hi Ian,
When you check out the remote feature branch do you tick 'Create New Branch'?

![Screenshot 2024-09-07 191351](https://github.com/user-attachments/assets/3d1b7a13-12b6-4140-8fdf-77280dec902b)

If I tick it, will have to tick the override option as well and will end up with the same screenshot I provided before.
If I don't tick it, I get this one:

![Screenshot 2024-09-07 191516](https://github.com/user-attachments/assets/9e704ed7-2ba4-4ca5-b591-c470205e6692)

Does it seen right to you?...

Thanks
Cara

**Reply by ian-tiggeman** [reply]
No, do not tick 'create new branch'.
Yes, I found that is how it looks with a remote branch checked out.

**Reply by ian-tiggeman** [reply]
Having said that, as I side note..., I don't have the 'origin/HEAD' branch... I am not sure where you got that from.
